tracer/test/Datadog.Trace.Tests/Telemetry/Collectors/ConfigurationTelemetryCollectorTests.cs

```diff
@@ -238,6 +239,36 @@ public void ConfigurationDataShouldReportDefaultValues()
         GetLatestValueFromConfig(data, "DD_TRACE_HEADER_TAGS", ConfigurationOrigins.Default).Should().Be(string.Empty);
         GetLatestValueFromConfig(data, "DD_LOGS_INJECTION", ConfigurationOrigins.Default).Should().Be(false);
         GetLatestValueFromConfig(data, "DD_TRACE_SAMPLE_RATE", ConfigurationOrigins.Default).Should().Be(1.0);
+        GetLatestValueFromConfig(data, "instrumentation_source", ConfigurationOrigins.Default).Should().Be("unknown");
+    }
+
+    [Fact]
+    public void ConfigurationDataShouldReportSSIValues()
+    {
+        // Store original variable
+        var originalInjectionEnabled = Environment.GetEnvironmentVariable("DD_INJECTION_ENABLED");
+        var originalInjectionForce = Environment.GetEnvironmentVariable("DD_INJECT_FORCE");
+        // Set environment variable to override
+        try
+        {
+            Environment.SetEnvironmentVariable("DD_INJECTION_ENABLED", "tracer");
+            Environment.SetEnvironmentVariable("DD_INJECT_FORCE", "true");
+
+            var collector = new ConfigurationTelemetry();
+            var source = new NameValueConfigurationSource(new NameValueCollection());
+            _ = new TracerSettings(source, collector, new OverrideErrorLog());
+            _ = new SecuritySettings(source, collector);
+            var data = collector.GetData();
+            GetLatestValueFromConfig(data, ConfigTelemetryData.SsiInjectionEnabled).Should().Be("tracer");
+            GetLatestValueFromConfig(data, ConfigTelemetryData.SsiAllowUnsupportedRuntimesEnabled).Should().Be("true");
+            GetLatestValueFromConfig(data, ConfigTelemetryData.InstrumentationSource).Should().Be("ssi");
+        }
+        finally
+        {
+            // Restore original variable
+            Environment.SetEnvironmentVariable("DD_INJECTION_ENABLED", originalInjectionEnabled);
+            Environment.SetEnvironmentVariable("DD_INJECT_FORCE", originalInjectionForce);
+        }
```

Comment:
We have an attribute you can set on the test class to do this for you, `[EnvironmentRestorer("DD_INJECTION_ENABLED", "DD_INJECT_FORCE")]` 🙂